substratevm/src/com.oracle.svm.jfr/src/com/oracle/svm/jfr/JfrBufferAccess.java

```diff
@@ -68,23 +68,24 @@ public static void free(JfrBuffer buffer) {
         ImageSingletons.lookup(UnmanagedMemorySupport.class).free(buffer);
     }
 
-    @Uninterruptible(reason = "Called from uninterruptible code.", mayBeInlined = true)
+    @Uninterruptible(reason = "Prevent safepoints as those could change the top pointer.", callerMustBe = true)
```

Comment:
I don't think that `callerMustBe = true` is absolutely necessary here (top is just set to a new value, it is not returned)